services/frontend-service/src/ui/components/ProductVersion/ProductVersion.tsx

```diff
@@ -15,40 +15,120 @@ along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>
 Copyright 2023 freiheit.com*/
 import './ProductVersion.scss';
 import * as React from 'react';
-import { refreshTags, useTags } from '../../utils/store';
+import { refreshTags, useTags, getSummary, useSummaryDisplay } from '../../utils/store';
+import { DisplayManifestLink, DisplaySourceLink } from '../../utils/Links';
+import { Spinner } from '../Spinner/Spinner';
+import { ProductSummary } from '../../../api/api';
 
 export type ProductVersionProps = {
     environment: string;
 };
+
 export const ProductVersion: React.FC<ProductVersionProps> = (props) => {
     React.useEffect(() => {
+        setShowTagsSpinner(true);
         refreshTags();
     }, []);
     const { environment } = props;
+    const summaryResponse = useSummaryDisplay();
     const [open, setOpen] = React.useState(false);
-    const openClose = React.useCallback(() => {
-        setOpen(!open);
-    }, [open, setOpen]);
-    const tags = useTags();
+    const openClose = React.useCallback(
+        (e: React.ChangeEvent<HTMLSelectElement>) => {
+            setShowSummarySpinner(true);
+            getSummary(e.target.value, environment);
+            setOpen(!open);
+        },
+        [open, setOpen, environment]
+    );
+    const [showTagsSpinner, setShowTagsSpinner] = React.useState(false);
+    const [showSummarySpinner, setShowSummarySpinner] = React.useState(false);
+    var versionToDisplay = (app: ProductSummary): string => {
+        if (app.displayVersion !== '') {
+            return app.displayVersion;
+        }
+        if (app.commitId !== '') {
+            return app.commitId;
+        }
+        return app.version;
+    };
+
+    const tagsResponse = useTags();
+    const [displaySummary, setDisplayVersion] = React.useState(false);
+
+    React.useEffect(() => {
+        if (tagsResponse.response.tagData.length > 0) {
+            setShowSummarySpinner(true);
+            getSummary(tagsResponse.response.tagData[0].commitId, environment);
```

Comment:
why are we calling `getSummary` twice?

Reply:
two separate instances, one is for when a user selects an option in the dropdown and the other is when the user first reaches the product version page so that results are loaded